### DISCUSSION TITLE
Person Detection - Ring Doorbell Camera Motion

### DISCUSSION THREAD

**Discussion by jjotto** [tsightler/ring-mqtt#670] [General Discussion]
![image](https://github.com/tsightler/ring-mqtt/assets/22684269/92e31936-82b7-41d7-9263-0af9295fa44d)

![image](https://github.com/tsightler/ring-mqtt/assets/22684269/ac64e32f-fcfe-4817-b3cc-f9aa75641c82)

I'm thinking I should be able to use this attribute to see if a motion event is detected as a person or just motion.  I may not be understanding it correctly.  I tried to use it as an event trigger for an automation, but I must be doing something wrong.  I also tried to make that attribute a sensor using a template just to see how that would work, but it seems to never change from "unknown" so it must not be setup right either.  Thoughts or suggestions.

**Comment by tsightler** [comment]
This attribute indicates whether the last motion event was a motion or person event based on Ring's Smart Alert feature, which requires a Ring Protect subscription and for Smart Alerts to be enabled for the camera.

I would not suggest using it as a trigger, but as a condition for any automation.  For example, if you want to send a notification only if a person is detected during the motion event you would create an automation that triggers on motion events, but uses the condition that the personDetected attribute is true, no more complicated than that.

**Reply by PetitBiscuit77** [reply]
Hello Tsightler,
Thank you for adding the attribute PersonDetected.
With HA I would like to have a notification when my camera detects a Person and no when a motion is detected, this feature works fine with my app Ring and I have a subscription Ring.

I try your solution with a trigger when a motion is detected and I add a condition with PersonDetected but it doesn't work.
In attacment my automation:
![image](https://github.com/tsightler/ring-mqtt/assets/129862176/a25f8618-1d14-4665-b056-818ab1e4b4af)

If I check my action, this isn't executed because my condition is false and I don't understand because my state seems to be OK.
![image](https://github.com/tsightler/ring-mqtt/assets/129862176/99299adc-f5fb-45de-8b1c-6c617b8c31ac)

I've got to do something wrong if you have a tips, i'll take it.

Thanks
PetitBiscuit77

**Reply by jjotto** [reply]
I've been using a different solution to get the person detection into home assistant.  It has been working well.  I was having problems getting it to work how I wanted with the addon.  If you have the subscription, you can try it too.

Use the Alexa app on your phone to make a new routine where you trigger a scene change when a person is detected.  You can then put that into home assistant as an entity.

Hope that helps!

![Screenshot_20231211-154545](https://github.com/tsightler/ring-mqtt/assets/22684269/d7b7d70f-e44b-4b2c-ab6b-f7a4a4ced670)
![Screenshot_20231211-154618](https://github.com/tsightler/ring-mqtt/assets/22684269/704bb2fa-bbe0-47ed-81e7-c91dbb37e5de)
![image](https://github.com/tsightler/ring-mqtt/assets/22684269/243ab3bc-a736-454f-b143-d974e3a3aa17)
![image](https://github.com/tsightler/ring-mqtt/assets/22684269/85e2e758-7350-4fbc-bd51-05b83d477575)

**Reply by tsightler** [reply]
This is really a Home Assistant question, not specific to the addon, and I generally steer clear of trying to help people with generic HA things because that's not my expertise, however, I think your case looks pretty clear, it appears that the UI is making a rule that attempts to match the string true while it should be a boolean match.  The attribute is not a string of 'true' or 'false' it is a boolean true/false.

Look at your screenshot, it's comparing `state: true` but `wanted_state: 'true'` the quotes means it's trying to match the string 'true' not the boolean true/false state.

**Reply by PetitBiscuit77** [reply]
Ohhh thank you I hadn't thought of using Alexa. So I tried but I don't have the person detected function under Alexa.
![image](https://github.com/tsightler/ring-mqtt/assets/129862176/2108affb-f102-4f8c-b6a1-a009d2bccdf1)
![image](https://github.com/tsightler/ring-mqtt/assets/129862176/cd2aa07f-0d2b-4e89-aebe-e1944b56a423).

my camera supports person detection well.

I think I understand how you use this , just a question when you use the scene "Person detected front door" in alexa it's the scene created with  Home Assistant 

Thant you

**Reply by tsightler** [reply]
One other quick note, Home Assistant UI editor for automations does not support boolean directly, however, you can just switch to yaml editor and remove the quotes around `'true'` and save it and it will work.

**Reply by jjotto** [reply]
Yes, I added a helper button like shown below:
![image](https://github.com/tsightler/ring-mqtt/assets/22684269/8e0c15a7-d63e-4aab-9c7c-fe8bb0e12df5)

**Reply by PetitBiscuit77** [reply]
Hi ,
The solution when I switch to YAML conf works if I delete "".
I think that is the best solution for the moment because Alexa's routine doesn't work for me.
I will test all of this in depth, if I find an improvments or detals,  I will post them here, This could help people for same use case
Thank you

**Comment by jjotto** [comment]
Thanks for the quick response!

Ok, that makes sense.  I have the subscription and the smart alerts (I only get notifications on my phone when a person is detected).  I'm going to tinker with it for a bit more and see if I can get everything to work how I want.  I'm trying to save an image of the motion only if it is a person.

**Comment by jjotto** [comment]
![image](https://github.com/tsightler/ring-mqtt/assets/22684269/5416ff16-208d-4f3a-86fe-565065cb96d3)

I'm not sure if all these should give the same image or not.  It seems like they sometimes do and sometimes don't.
Often it seems to grab the previous motion snapshot.

**Reply by tsightler** [reply]
You may need to add a short delay, the ring-mqtt logs are very detailed and log both when motion event is detected, when the snapshot is requested, and when it successfully retrieved the snapshot.  It's possible that it is failing in some cases in which case you would get the previous snapshot or that the timing is off slightly since it sometimes takes several hundred milliseconds after the notification for the snapshot to be retrieved.

In any case, between the ring-mqtt logs and the automation traces, you should be able to get a very clear picture of exactly what is happening for every event.

**Comment by alvinchen1** [comment]
Sorry to revive an old topic, but I noticed this attribute and also wanted to do something interesting with it similar to what OP was thinking, however I think there is a small bug in the attribute.  I used a template just like what OP did for each of my four cameras:

<img width="531" alt="image" src="https://github.com/tsightler/ring-mqtt/assets/26409535/4d361bb1-dc2c-4c18-ad73-dbcf817180c8">

I have each camera set for Smart Alerts

![Screenshot_20240115-060136](https://github.com/tsightler/ring-mqtt/assets/26409535/9f9f80de-a727-4df3-9b00-04c946640aff)

And in the Ring app, Person detected is working

![Screenshot_20240115-060156](https://github.com/tsightler/ring-mqtt/assets/26409535/844389ea-55a9-4c9b-bc8f-a43cdbc91cb6)

The bug I noticed is that once Person detected is triggered, it DOES change the attribute from false to true, but it never resets to false until I restart the add-on.

<img width="899" alt="image" src="https://github.com/tsightler/ring-mqtt/assets/26409535/4248deec-5565-4821-a2b8-a1275286a854">

In the History graph above, I have the new sensor (person detected), followed by the camera's corresponding motion detected graph.  Once I restart the add-on, reboot the camera or reboot the entire system, person detected is reset to false until the next time person detected is truly triggered.  You can see from the history graph that the attribute changes from false to true with the first motion detection becoming true, and remains true even when motion detection becomes false.

**Reply by kmanan** [reply]
Sorry for reviving an old thread but Google searches got me here, I'm trying to use this as a trigger for lights (have Ring pro) but the sensor is always "clear" in HA even though I am able to get Ring and Alexa announce a person. Am I using the wrong sensor? There are 10 motion sensors for a doorbell... ![image](https://github.com/user-attachments/assets/611f0338-c744-465f-92e1-50479ba0e947)

**Reply by tsightler** [reply]
I would be preferable not to revive old threads, especially since the screenshot you are showing has nothing to do with motion detection at all, so it just creates confusion.  That dropdown is the event selector, it allows you to select which event will be shown when the even camera is viewed as is explained in the documentation.  You looking for the motion sensor.

Everything that could ever be written about sensors is pretty much already covered in the wiki or in zillions of other threads here, there's nothing new or changed in ages about how it works, how to troubleshoot it, etc.

**Reply by kmanan** [reply]
Thanks, makes sense. None of my ring MQTT device are registering motion, was hoping this would be the easy fix. Thanks though, I'll keep searching for a solution

**Reply by tsightler** [reply]
You don't have to search, the troubleshooting steps are well documented in the support and troubleshooting section of the wiki.  If you follow those steps, you should be able to get it working as the steps are outlined there.

**Comment by b5turbo** [comment]
I'm also sorry for reviving and old thread but nobody answered the supposed bug reported by @alvinchen1. I would like to explain there is no bug and the specifically configured Ring app notification settings are causing the reported behaviour. 

The 'personDetected' attribute is set when the last motion alert detected a person. Once the attribute is set to true, it won't change until the next motion alert doesn't detect a person.

The Ring app configuration shown by @alvinchen1 only sends alerts when a person is detected. Therefore once set, it will always remain true. Other motion alert types must be enabled for the 'personDetected' attribute to be set back to false.
